src/main/java/com/artipie/asto/s3/S3Storage.java

```diff
@@ -173,22 +157,16 @@ public S3Storage(final S3AsyncClient client, final String bucket, final boolean
         final CompletableFuture<Void> result;
         final Content onetime = new Content.OneTime(content);
         if (this.multipart) {
-            result = complementWithSize(onetime, S3Storage.MIN_MULTIPART).thenCompose(
-                updated -> {
-                    final CompletableFuture<Void> future;
-                    final Optional<Long> size = updated.size();
-                    if (size.isPresent() && size.get() < S3Storage.MIN_MULTIPART) {
-                        future = this.put(key, updated);
-                    } else {
-                        future = this.uploadMultipart(key, updated);
-                    }
-                    return future;
-                }
-            );
+            result =
+                new EstimatedContentCompliment(onetime, S3Storage.MIN_MULTIPART)
+                    .estimate()
+                    .toCompletableFuture()
```

Comment:
@andreoss this call is not needed, `thenCompose` is a part of `CompletionStage` interface

Reply:
@g4s8 Done